frontend/src/features/map_planning/layers/shade/components/ShadingAttributeEditForm.tsx

```diff
@@ -0,0 +1,342 @@
+      executeAction(
+        new UpdateShadingRemoveDateAction({
+          removeDate: removeDate,
+          id: shading.id,
+        }),
+      );
+    });
+  };
+
+  const onShadeChange = ({ shade }: ShadingDateAttribute) => {
+    if (!shade) return;
+
+    shadings.forEach((shading) => {
+      executeAction(
+        new UpdateShadingAction({
+          shade: shade,
+          id: shading.id,
+          geometry: shading.geometry,
+        }),
+      );
+    });
+  };
+
+  const onDeleteShading = () => {
+    selectShadings(null);
+    shadings.forEach((shading) => {
+      executeAction(new DeleteShadingAction(shading));
+    });
+  };
+
+  const getCommonShade = () => {
+    const commonShade = shadings[0].shade;
+    for (const shading of shadings) {
+      if (shading.shade != commonShade) return undefined;
+    }
+
+    return commonShade;
+  };
+
+  const getCommonAddDate = () => {
+    const commonAddDate = shadings[0].addDate;
+    for (const shading of shadings) {
+      if (shading.addDate != commonAddDate) return undefined;
+    }
+
+    return commonAddDate;
+  };
+
+  const getCommonRemoveDate = () => {
+    const commonAddDate = shadings[0].removeDate;
+    for (const shading of shadings) {
+      if (shading.removeDate != commonAddDate) return undefined;
+    }
+
+    return commonAddDate;
+  };
+
+  return (
+    <ShadingAttributeEditForm
+      showPolygonTools={false}
+      shade={getCommonShade()}
+      addDate={getCommonAddDate()}
+      removeDate={getCommonRemoveDate()}
+      onAddDateChange={onAddDateChange}
+      onRemoveDateChange={onRemoveDateChange}
+      onShadeChange={onShadeChange}
+      onDeleteShading={onDeleteShading}
+    />
+  );
+}
+
+function ShadingAttributeEditForm({
+  onAddDateChange,
+  onDeleteShading,
+  onRemoveDateChange,
+  onShadeChange,
+  showPolygonTools,
+  addDate,
+  removeDate,
+  shade,
+}: ShadingAttributeEditFormProps) {
+  const { t } = useTranslation('shadeLayer');
+  const isReadOnlyMode = useIsReadOnlyMode();
+
+  const formFunctions = useForm<ShadingDateAttribute>({
+    defaultValues: {
+      addDate,
+      removeDate,
+    },
+    resolver: zodResolver(ShadingAttributeEditFormSchema),
+  });
+
+  const shadeOptions: Array<SelectOption> = [
+    { value: Shade.LightShade, label: t('shading_amount.light_shade') },
+    { value: Shade.PartialShade, label: t('shading_amount.partial_shade') },
+    { value: Shade.PermanentShade, label: t('shading_amount.permanent_shade') },
+    { value: Shade.PermanentDeepShade, label: t('shading_amount.permanent_deep_shade') },
+  ];
+
+  useEffect(() => {
+    formFunctions.setValue('addDate', addDate, {
+      shouldDirty: false,
+      shouldTouch: false,
+    });
+    formFunctions.setValue('removeDate', removeDate, {
+      shouldDirty: false,
+      shouldTouch: false,
+    });
+    formFunctions.setValue('shade', shade);
+  }, [addDate, removeDate, shade]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Please explain why you used `// eslint-disable-line react-hooks/exhaustive-deps`.
I guess you can also include `formFunctions.setValue` as a dep.
No advantages that I can think of.